### PR TITLE
chore(ct): throw error when props are not json serializable

### PR DESCRIPTION
**Pull request by sand4rt**
Related to: https://github.com/microsoft/playwright/pull/21400, https://github.com/microsoft/playwright/issues/22003

### CODE REVIEW

File: packages/playwright-test/src/mount.ts [hunk 1]
```diff
@@ -72,6 +74,24 @@ export const fixtures: Fixtures<
     },
   };
 
+const jsonType: Record<string, Function> = {
+  string: (value: any) => typeof value === 'string',
+  number: (value: any) => typeof value === 'number',
+  boolean: (value: any) => typeof value === 'boolean',
+  null: (value: any) => value === null,
+  array: (value: any) => Array.isArray(value) && value.every(isJson),
+  object: (value: any) => typeof value === 'object' && value !== null && !Array.isArray(value)
```
Comment (pavelfeldman) [inline packages/playwright-test/src/mount.ts:83]: JSON will have a default serializing conventions for types like Date, but I guess it is Ok to be more restrictive here.

**Review by pavelfeldman — APPROVED** [review]
Fix those nits and we are good.

**Comment by sand4rt** [comment]
> Fix those nits and we are good.

Done!

**Comment by sand4rt** [comment]
> Fix those nits and we are good.

Done, also added the check for `await component.update({ props: { boop: 'boop' }})` : https://github.com/microsoft/playwright/pull/22025/files#diff-f6515604ba6e73e5ba509d2d713faacc6e09f0a06b1db6ca30d0ca7c6ffb4e55R68

File: packages/playwright-test/src/mount.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/playwright-test/src/mount.ts:86]: Look at what `JSON.stringify({ data: new Date() })` produces, but don't bother.

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**